aiida/repository/backend/sandbox.py

```diff
@@ -16,11 +15,12 @@ class SandboxRepositoryBackend(AbstractRepositoryBackend):
     """Implementation of the ``AbstractRepositoryBackend`` using a sandbox folder on disk as the backend."""
 
     def __init__(self):
-        self._sandbox = None
+        from aiida.common.folders import SandboxFolder
+        self._sandbox: typing.Optional[SandboxFolder] = None
 
     def __str__(self) -> str:
         """Return the string representation of this repository."""
-        return f'SandboxRepository: {self._sandbox.abspath}'
+        return f'SandboxRepository: {self._sandbox.abspath if self._sandbox else "null"}'
```

Comment:
This was clearly a bug and not caught by the tests. I think it would be good to add a quick test, both for the sandbox and diskobject backends.

Reply:
done; I adapted them slightly, to return `SandboxRepository: <uninitialised>`/`DiskObjectStoreRepository: <uninitialised>` before initialisation